File: types/digest.ts
export interface Digest {
  id: string
  transcript: string
  summary: string
  overview: string
  key_decisions: string[]
  action_items: string[]
  created_at: string
  public_id?: string
}

export interface CreateDigestRequest {
  transcript: string
}

export interface CreateDigestResponse {
  digest: Digest
}

export interface ListDigestsResponse {
  digests: Digest[]
}

File: backend/src/models/digest.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime


class TranscriptRequest(BaseModel):
    transcript: str


class DigestResponse(BaseModel):
    id: str
    overview: str
    key_decisions: List[str]
    action_items: List[str]
    created_at: datetime
    public_id: Optional[str] = None